platforms/platforms.go

```diff
@@ -0,0 +1,129 @@
+package platforms
+
+import (
+	"regexp"
+	"runtime"
+	"strings"
+
+	"github.com/containerd/containerd/errdefs"
+	specs "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+)
+
+type platformKey struct {
+	os      string
+	arch    string
+	variant string
+}
+
+var (
+	selectorRe = regexp.MustCompile(`^[A-Za-z0-9_-]$`)
+)
+
+// ParseSelector parses the platform selector syntax into a platform
+// declaration.
+//
+// Platform selectors are in the format <os|arch>[/<arch>[/<variant>]]. The minimum
+// required information for a platform selector is the operating system.
+//
+// The minimum required information for a platform selector is the operating
+// system or architecture. If there is only a single string (no slashes(, the
```

Comment:
One comment says the minimum requirement for a platform selector is the operating system, but the next paragraph says operating system or architecture, which matches the format specification. Remove the incorrect sentence and probably combine the two paragraphs into one.